Hudl.Mjolnir/Config/MjolnirConfigurationObserver.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+
+namespace Hudl.Mjolnir.Config
+{
+    internal sealed class MjolnirConfigurationObserver<T> : IObserver<MjolnirConfiguration>
+    {
+        private T _currentValue;
+        private readonly Func<MjolnirConfiguration, T> _expression;
+        private readonly Action<T> _onChange;
+        internal MjolnirConfigurationObserver(MjolnirConfiguration currentConfig, 
+            Func<MjolnirConfiguration, T> propertyToCheck, Action<T> onChange)
+        {
+            _expression = propertyToCheck;
+            _currentValue = _expression(currentConfig);
+            _onChange = onChange;
+        }
+
+        public void OnCompleted()
+        {
+            throw new NotImplementedException();
+        }
+
+        public void OnError(Exception error)
+        {
+            throw new NotImplementedException();
```

Comment:
Is this the best approach here? i.e. is it impossible for `OnError` to happen, or should we have better handling here?

Reply:
Good find. I'll change to no-op.
FWIW With my implementation, it's not possible for OnError or OnCompleted to be fired.